app/controllers/v0/messages_controller.rb

```diff
@@ -5,7 +5,7 @@ class MessagesController < SMController
     include Filterable
 
     def index
-      resource = client.get_folder_messages(@current_user.uuid, params[:folder_id].to_s, use_cache?)
+      resource = client.get_folder_messages(@current_user.uuid, params[:folder_id].to_s)
```

Comment:
`get_folder_messages` will now determine whether to use cache or not based on if the length of our cached_messages === total folder count or not. See below.